### DISCUSSION TITLE
Display Github PR files as a source

### DISCUSSION THREAD

**Discussion by ogirginc** [nvim-neo-tree/neo-tree.nvim#611] [Q&A]
Similar to `git_status`, I want to display the files on a GitHub PR as a tree. I have tried adding[1] it as a custom source, but failed. Is there some detailed guide or a blog post where I can read and learn?

What I want is basically showing the outcome of the line above as a tree:

```sh
gh pr view $(git branch --show-current) --json files | jq ".files[].path"
```

My goal is to mimic[1] https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github for Neovim.

Thank you in advance!

[1] https://github.com/nvim-neo-tree/example-source
[2] https://www.reddit.com/r/neovim/comments/ys879l/github_pull_requests_and_issues_plugin_for_neovim/

**Comment by cseickel** [comment]
You can already do this with the git_status source. If you checkout the PR, you can then use:

```
:Neotree git_status main
```

Replace `main` with whatever branch you want to diff against. This is the [git_base](https://github.com/nvim-neo-tree/neo-tree.nvim#git_base) option in the command. This will show you all changes made between your current branch and the `git_base`.

I realize documentation is an area where Neo-tree is lacking. Sorry about that!

**Reply by ogirginc** [reply]
Unfortunately not. For example, a PR can have multiple merge commits from master/main to rebase. When I have used git_base, I did get those files too, which makes the feature useless in my case. 😕

That’s why I want to create a new source, where I can ignore commits for rebasing. The gh command I share achieves this. Not sure if it is possible without gh dependency.

Do you still think it is possible?

**Reply by cseickel** [reply]
It's definitely possible. I'll take a look later and give you some more detailed advice.

Have you created a repo that has your work up until this point?

**Reply by ogirginc** [reply]
I don’t have the code unfortunately. It was more like a scratch pad and I have deleted it, after I realised it’s not working. 😕
